var/spack/repos/builtin/packages/flux-sched/package.py

```diff
@@ -35,6 +36,11 @@ class FluxSched(AutotoolsPackage):
     variant('cuda', default=False, description='Build dependencies with support for CUDA')
 
     depends_on("boost+graph@1.53.0,1.59.0:")
+
```

Comment:
@trws : happy to help, but I think I need some clarification on what you'd like me to test.  We have the `+graph` variant defined above.  I think the package has been like that for a while.  That should be sufficient, right?

Actually, now that I'm thinking about it, the new data-staging plugin requires `filesystem`, so that's one change that needs to be made.  But besides that, I'm a little lost as to what behavioral changes this PR makes and what needs to be tested on the Flux-side as a consequence.

Reply:
The boost package will no longer have a default set of stuff that gets built in addition to the selected variants, so anything that gets used but isn't listed because its variant didn't exist will cause issues.  Basically, if it builds with `boost+graph+filesystem` after this PR, it's all good, but it might not if something that used to be included by default is no longer there.

Reply:
Makes sense.  Sorry, I just now am seeing the context at the top of this PR.  I just saw your issue that I was tagged in and went straight to this diff.  My bad.

I'll definitely add it to my TODO list.  No promises on a timeline.  Probably will get around to it shortly after the next monthly Flux release when I drop the next set of release numbers and checksum.

Reply:
I applied this patch to the current PR branch: 

```diff
diff --git a/var/spack/repos/builtin/packages/flux-sched/package.py b/var/spack/repos/builtin/packages/flux-sched/package.py
index 41781aa232..e2bbff501a 100644
--- a/var/spack/repos/builtin/packages/flux-sched/package.py
+++ b/var/spack/repos/builtin/packages/flux-sched/package.py
@@ -36,12 +36,10 @@ class FluxSched(AutotoolsPackage):
     variant('cuda', default=False, description='Build dependencies with support for CUDA')

     depends_on("boost+graph@1.53.0,1.59.0:")
+    depends_on("boost+filesystem", when="@0.13.0:")

-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
     depends_on("py-pyyaml")
+    depends_on("py-jsonschema", when="@0.8.0:")
     depends_on("libxml2@2.9.1:")
     depends_on("yaml-cpp")
     depends_on("uuid")
```

I can concretize flux-sched with the proper boost variants when I provide an explicit version for flux-sched:
```
herbein1@quartz1922 ~/Repositories/spack (boost !?)
❯ spack spec flux-sched@0.17.0                                                                          12:20:01 ()
Input spec
--------------------------------
flux-sched@0.17.0
                                                                                                                    Concretized
--------------------------------
flux-sched@0.17.0%gcc@4.9.3~cuda arch=linux-rhel7-x86_64
    ^boost@1.76.0%gcc@4.9.3~atomic~chrono~clanglibcpp~container~context~coroutine~date_time~debug~exception~fiber+fi
lesystem+graph~icu~iostreams~locale~log~math~mpi+multithreaded~numpy~pic~program_options~python~random~regex~seriali
zation+shared~signals~singlethreaded~system~taggedlayout~test~thread~timer~versionedlayout~wave cxxstd=98 visibility
=hidden arch=linux-rhel7-x86_64
<snip>
```

It also works for older flux-sched's that don't require `boost+filesystem`:
```
herbein1@quartz1922 ~/Repositories/spack (boost !?)
❯ spack spec flux-sched@0.12.0                                                                          12:20:44 ()
Input spec
--------------------------------
flux-sched@0.12.0

Concretized
--------------------------------
flux-sched@0.12.0%gcc@4.9.3~cuda arch=linux-rhel7-x86_64
    ^boost@1.76.0%gcc@4.9.3~atomic~chrono~clanglibcpp~container~context~coroutine~date_time~debug~exception~fiber~fi
lesystem+graph~icu~iostreams~locale~log~math~mpi+multithreaded~numpy~pic~program_options~python~random~regex~seriali
zation+shared~signals~singlethreaded~system~taggedlayout~test~thread~timer~versionedlayout~wave cxxstd=98 visibility
=hidden arch=linux-rhel7-x86_64 
```

But if you don't provide an explicit version for flux-sched, everything borks:
```
9s herbein1@quartz1922 ~/Repositories/spack (boost !?)
❯ spack spec flux-sched                                                                                 12:20:30 ()
Input spec
--------------------------------
flux-sched

Concretized
--------------------------------
==> Error: An unsatisfiable variant constraint has been detected for spec:
                                                                                                                        boost@1.76.0%gcc@4.9.3~atomic~chrono~clanglibcpp~container~context~coroutine~date_time~debug~exception~fiber~fil
esystem+graph~icu~iostreams~locale~log~math~mpi+multithreaded~numpy~pic~program_options~python~random~regex~serializ
ation+shared~signals~singlethreaded~system~taggedlayout~test~thread~timer~versionedlayout~wave cxxstd=98 visibility=
hidden arch=linux-rhel7-x86_64


while trying to concretize the partial spec:
                                                                                                                        flux-sched@0.17.0%gcc@4.9.3~cuda arch=linux-rhel7-x86_64


flux-sched requires boost variant +filesystem, but spec asked for ~filesystem
```

Not sure why it is saying that the spec is asking for `~filesystem`.  Any thoughts/advice?